### PR TITLE
38680

### PR DESCRIPTION
**Pull request by nks2109**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [38680](https://devops.dchbx.org/redmine/issues/38680) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | n/a |
| Peer Review Story Points | n/a |
| Ticket Author | Maurice Wiggins |

### CODE REVIEW

File: lib/tasks/update_aptc_credits.rake [hunk 1]
```diff
@@ -1,7 +1,10 @@
 require File.join(Rails.root,"app","data_migrations","update_aptc_credits.rb")
 
   # This rake task merges people in Glue.
-  # format RAILS_ENV=production bundle exec rake migrations:update_aptc_credits policy_id="304676" aptc="0" start_on="1/1/2018" end_on="9/30/2018" delete_credit="true"
+  #Below rake is to delete aptc credit
+    # format RAILS_ENV=production bundle exec rake migrations:update_aptc_credits policy_id="304676" aptc="0" start_on="1/1/2018" end_on="9/30/2018" delete_credit="true"
+  #Below rake is to create a new aptc credit 
+    # format RAILS_ENV=production bundle exec rake migrations:update_aptc_credits policy_id="304676" aptc="200" start_on="1/1/2018" end_on="9/30/2018" pre_amt_tot="300" tot_res_amt="100" delete_credit="nil"
```
Comment (ZASMan) [inline lib/tasks/update_aptc_credits.rake]: Can we get rid of the requirement of adding the `delete_credit` argument if its for creating a new APTC credit? seems excessive

File: lib/tasks/update_aptc_credits.rake [hunk 2]
```diff
@@ -1,7 +1,10 @@
 require File.join(Rails.root,"app","data_migrations","update_aptc_credits.rb")
 
   # This rake task merges people in Glue.
-  # format RAILS_ENV=production bundle exec rake migrations:update_aptc_credits policy_id="304676" aptc="0" start_on="1/1/2018" end_on="9/30/2018" delete_credit="true"
+  #Below rake is to delete aptc credit
```
Comment (ZASMan) [inline lib/tasks/update_aptc_credits.rake]: Add a space in the comment here for consistency?

File: spec/data_migrations/update_aptc_credits_spec.rb [hunk 3]
```diff
@@ -54,15 +55,16 @@
       policy.reload
       credit.reload
       
-      expect(credit.pre_amt_tot).to eq 200
+      expect(credit.pre_amt_tot).to eq 300
       expect(credit.tot_res_amt).to eq 100
-      expect(credit.end_on).to eq Date.new(2018,7,2)
+      expect(credit.end_on).to eq Date.new(2019,2,28)
```
Comment (ZASMan) [inline spec/data_migrations/update_aptc_credits_spec.rb]: Is it part of our style to add spaces after the numbers here?

File: spec/data_migrations/update_aptc_credits_spec.rb [hunk 4]
```diff
@@ -81,36 +83,64 @@
       policy.reload
       credit.reload
 
-      expect(credit.pre_amt_tot).to eq 200
-      expect(credit.tot_res_amt).to eq 150
-      expect(credit.end_on).to eq Date.new(2018,5,2)
-      expect(credit.start_on).to eq Date.new(2018,4,2)
+      expect(credit.pre_amt_tot).to eq 300
+      expect(credit.tot_res_amt).to eq 100
+      expect(credit.end_on).to eq Date.new(2019,2,28)
```
Comment (ZASMan) [inline spec/data_migrations/update_aptc_credits_spec.rb]: Same as above about spaces if necessary

File: spec/data_migrations/update_aptc_credits_spec.rb [hunk 5]
```diff
@@ -81,36 +83,64 @@
...
+      expect(policy.applied_aptc).to eq 100
       expect(policy.tot_res_amt).to eq policy.pre_amt_tot - policy.applied_aptc
 
     end
   end
 end
 
+describe "Creating new aptc credits", dbclean: :after_each do 
+  let(:given_task_name) { "update_aptc_credits" }
+  let(:policy) { FactoryGirl.create(:policy, applied_aptc:"250", pre_amt_tot:"350",tot_res_amt:"100") }
```
Comment (ZASMan) [inline spec/data_migrations/update_aptc_credits_spec.rb]: Spaces after colons needed?

File: lib/tasks/update_aptc_credits.rake (hunk 1 above)
Comment (nks2109) [reply lib/tasks/update_aptc_credits.rake]: delete_credit is environmental variable which is required to delete the credit. If we are not deleting credit then an argument should be passed to skip that condition

**Pull request merged**